SoftLayer/CLI/hardware/detail.py

```diff
@@ -107,6 +108,46 @@ def cli(env, identifier, passwords, price):
             pass_table.add_row([item['username'], item['password']])
         table.add_row(['remote users', pass_table])
 
+    if components:
+
+        _mask = 'id,hardwareComponentModel[name,' \
```

Comment:
moving `_mask` and `_filter` logic within the `getcomponents` , will look better.